packages/metal-quartz-link/src/MetalQuartzLink.soy

```diff
@@ -0,0 +1,56 @@
+{namespace MetalQuartzLink}
+
+/**
+ * This renders an anchor element.
+ */
+{template .render}
+	{@param? dataToggle: string}
+	{@param? elementClasses: string}
+	{@param? href: string}
+	{@param? html: html}
+	{@param? icon: ?}
+	{@param? label: html}
+	{@param? id: string}
+
+	{if $html}
+		{$html}
+	{else}
+		{let $attrs kind="attributes"}
+			{if $elementClasses}
+				class="{$elementClasses}"
+			{/if}
+
+			{if $dataToggle}
+				data-toggle="{$dataToggle}"
+			{/if}
+
+			{if $href}
+				href="{$href}"
+			{/if}
+
+			{if $id}
+				id="{$id}"
+			{/if}
+		{/let}
+
+		<a {$attrs}>
+			{if $icon and $icon.alignment == 'left'}
+				{call MetalQuartzIcon.render data="all"}
+					{param elementClasses: $icon.elementClasses /}
+					{param spritemap: $icon.spritemap /}
+					{param symbol: $icon.symbol /}
+				{/call}
+			{/if}
+
+			{$label}
+
+			{if $icon and not $icon.alignment or $icon and $icon.alignment == 'right'}
+				{call MetalQuartzIcon.render data="all"}
```

Comment:
Let's pull this into another template in this file so we don't have to write it twice.

```
{template .icon}
  {call MetalQuartzIcon.render data="all"}
  ...
{/template}
```